### DISCUSSION TITLE
Webp export

### DISCUSSION THREAD

**Discussion by IsotoxalDev** [Orama-Interactive/Pixelorama#746] [Ideas]
## Why when GIF is there?
GIF is great and all but when it comes to exporting faster GIFS say for example 60 FPS. it won't work as it is the [standard behaviour](https://www.biphelps.com/blog/The-Fastest-GIF-Does-Not-Exist?source=techstories.org). anything over 50 fps. ie a GIF with delay less than 20 ms is by default exported as a 10 FPS Animation. And users expecting a fast animation would be disappointed. Hence the suggestion of webp.

## Why webp?
WebP is a modern image format that provides superior lossless and lossy compression for images, and is opensource and maintained by google. [More details regarding why webp is an apt solution](https://developers.google.com/speed/webp/faq#why_should_i_use_animated_webp)

**Comment by poyribn** [comment]
WEBP is not just a better GIF replacement.
WEBP wins against PNG for the same reason: same quality images in smaller files.

APNG is also a better format than GIF, but it is pointless because barely anything supports APNG.
WEBP is a different story: just like PNG and GIF, there is almost no software that does Not support WEBP.

this is why i want to export static images as WEBP too — instead of as PNG.

because of WEBP format's advantages, i think that replacing the PNG and GIF exports with WEBP — instead of having WEBP as a choice next to PNG and GIF — would be no real loss

**Comment by poyribn** [comment]
i found out by accident that i can drag-and-drop static — but not animated — WEBP images into Pixelorama, and they will import correctly.
this makes it much more odd that Pixelorama can't export static WEBP images

**Reply by IsotoxalDev** [reply]
Godot supports the import of WEBP natively, while there is no support for export. Even gif isn't natively supported, so we'd have to make a custom encoder, probably in gdscript which would be slow as we can see with our current gif export.

**Reply by poyribn** [reply]
thanks, i just found out that Godot 3.x doesn't support saving WEBP images out of the box.

the last time i used Godot was with Godot 4.0 beta, and its 'Image' node has the [save_webp()](https://web.archive.org/web/20230111091838/https://docs.godotengine.org/en/latest/classes/class_image.html#class-image-method-save-webp) function.

the newest Godot 4.0 version is [beta 11](https://godotengine.org/article/dev-snapshot-godot-4-0-beta-11), so we can't use Godot 4.0 for Pixelorama yet

**Reply by poyribn** [reply]
[Godot 4.0 is now stable](https://godotengine.org/article/godot-4-0-sets-sail/), so we can safely make Pixelorama in Godot 4.

Also, [DevDuck on YouTube has some tips for porting to Godot 4](https://www.youtube.com/watch?v=uT43-s8RGkA):
- if you're overwhelmed from too many error messages, run the game and fix the first error you find
- seeing the working code in Godot 3 is extremely helpful for fixing new bugs that appear in Godot 4

**Reply by Variable-ind** [reply]
> [Godot 4.0 is now stable](https://godotengine.org/article/godot-4-0-sets-sail/), so we can safely make Pixelorama in Godot 4.
> 
> Also, [DevDuck on YouTube has some tips for porting to Godot 4](https://www.youtube.com/watch?v=uT43-s8RGkA):
> 
>     * if you're overwhelmed from too many error messages, run the game and fix the first error you find
> 
>     * seeing the working code in Godot 3 is extremely helpful for fixing new bugs that appear in Godot 4

Pixelorama is probably planned to move to Godot 4 when Godot 4.1 gets released (there are still some things in 4.0 that need to be fixed/added)

**Comment by dnyrw** [comment]
``

**Reply by poyribn** [reply]
> ``

i guess you commented this by accident. please edit this into something you actually want to comment, or delete this comment.

if this comment has a meaning, please explain it.

**Comment by OverloadedOrama** [comment]
Now that we're using Godot 4, I added support for static webp and jpeg exporting. I'm leaving this discussion open though as exporting animated webp files is still not possible.
